api/entry.mli

```diff
@@ -20,7 +20,7 @@ type entry =
   (** Definition (possibly opaque). *)
   | Def   of loc * ident * is_opaque * term option * term
   (** Reduction rules declaration. *)
-  | Rules of loc * Rule.untyped_rule list
+  | Rules of loc * Rule.part_typed_rule list
```

Comment:
I don't think the name is informative enough. Maybe use the word `annotated` instead?